src/transformers/models/falcon_mamba/modeling_falcon_mamba.py

```diff
@@ -733,6 +757,10 @@ def prepare_inputs_for_generation(
                 )
             if cache_position[0] > 0:
                 input_ids = input_ids[:, -1].unsqueeze(-1)
+
+                if attention_mask is not None:
+                    attention_mask = attention_mask[:, -1].unsqueeze(-1)
+
```

Comment:
does this work if you generate with `inputs_embeds` as starter instead of `input_ids`?

Reply:
Yes I think so as we will always want to get the last value of the attention mask for the last token 🤔

Reply:
Just tried it with https://github.com/huggingface/transformers/pull/32677/commits/8978e00919cec6f72c874a9762d25c292ea08d80 and I can confirm it works !

Reply:
great, thanks!